qiskit/synthesis/two_qubit/two_qubit_decompose.py

```diff
@@ -96,7 +95,7 @@ def decompose_two_qubit_product_gate(special_unitary_matrix: np.ndarray):
 _id = np.array([[1, 0], [0, 1]], dtype=complex)
 
 
-class TwoQubitWeylDecomposition:
+class TwoQubitWeylDecomposition(two_qubit_decompose.TwoQubitWeylDecomposition):
```

Comment:
My original thinking for this was I wanted to drop the python side class altogether but it was going to be annoying to handle the `circuits()`, `from_bytes()`, and string/repr methods rust side, so this was what I settled on. I mostly did it this way to avoid duplication on the python side, but I think you're right I'm concerned about the overhead of copying the arrays for the getter on the `.K*` attributes. As cool as I thought it was to get subclassing working like this, I think you're right it'll be better to just wrap the inner rust object here, even if it ends up being a bit more verbose.

Reply:
I adjusted this in: https://github.com/Qiskit/qiskit/pull/11946/commits/a2fcc58a9edd2c1b1253d88a33d8054d0eca47f7 so we're wrapping the inner rust object now instead of subclassing. That's just a better pattern in this case.